src/services/ai-service.js

```
@@ -198,6 +208,12 @@ class AIService {
   
   // Select best available provider
   selectProvider(options, excludeName = null) {
+    const preferredKey = memoryState.get('ai.provider') || 'hf';
+    const preferred = this.providers.get(preferredKey);
```

Biome (2.1.2)

[error] This variable is used before its declaration.

The variable is declared here:


(lint/correctness/noInvalidUseBeforeDeclaration)